doltpy/sql/sync/oracle.py

```diff
@@ -0,0 +1,74 @@
+from sqlalchemy .engine import Engine
+from sqlalchemy import Table, select
+from doltpy.sql.sync.db_tools import (DoltAsSourceWriter,
+                                      drop_primary_keys,
+                                      DoltAsSourceUpdate)
+from doltpy.sql.write import hash_row_els
+from typing import List
+from sqlalchemy import MetaData, bindparam
+from copy import deepcopy
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def get_target_writer(engine: Engine, update_on_duplicate: bool = True) -> DoltAsSourceWriter:
```

Comment:
do you need singledispatch on these?

Reply:
What do you mean by singledispatch, as in a very similar function per database implementation (Oracle, PG, MySQL)?

Reply:
1 function signature mapping to different arguments or argument types. not sure why someone would want to use multiple at the same time, but if you use the singledispatch decorator you wouldn't have to worry about collisions

Reply:
documentation?

Reply:
@max-hoffman re-upping

Reply:
https://docs.python.org/3/library/functools.html#functools.singledispatch
probably just punt